### PR TITLE
Add LeanPipe contracts

### PR DESCRIPTION
**Pull request by mishioo**
Contracts as worked out in https://github.com/leancodepl/LeanPipe-Draft/pull/1

### CODE REVIEW

File: src/LeanCode.Contracts/NotificationEnvelope.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+namespace LeanCode.Contracts;
+
+public sealed class NotificationEnvelope
```
Comment (jakubfijalkowski) [inline src/LeanCode.Contracts/NotificationEnvelope.cs:3]: I miss ctor here. :)

Comment (mishioo) [reply src/LeanCode.Contracts/NotificationEnvelope.cs:3]: Public or restricted? With explicitly passed ID but types still interfered, I suppose? Do we want to have some type safety with reflection there?

Comment (mishioo) [reply src/LeanCode.Contracts/NotificationEnvelope.cs:3]: I've added a simple one without types check, let me know if this is what you've missed. :)

**Review by jakubfijalkowski — APPROVED** [review]

File: src/LeanCode.Contracts/NotificationEnvelope.cs (hunk 1 above)
Comment (jakubfijalkowski) [reply src/LeanCode.Contracts/NotificationEnvelope.cs:3]: That's exactly what I had in mind. :)

**Review by Dragemil — APPROVED** [review]

**Pull request merged**